### PR TITLE
Add state-sync limit to bor config

### PR DESCRIPTION
**Pull request by 0xKrishna**

### CODE REVIEW

File: consensus/bor/rest.go [hunk 1]
```diff
@@ -47,7 +47,7 @@ func NewHeimdallClient(urlString string) (*HeimdallClient, error) {
 	return h, nil
 }
 
-func (h *HeimdallClient) FetchStateSyncEvents(fromID uint64, to int64) ([]*EventRecordWithTime, error) {
+func (h *HeimdallClient) FetchStateSyncEvents(fromID uint64, to int64, totalLimit int) ([]*EventRecordWithTime, error) {
 	eventRecords := make([]*EventRecordWithTime, 0)
```
[the remark this answers is not in the document]
Comment (0xKrishna) [reply consensus/bor/rest.go:51]: But as per the current flow, we will have totalLimit = -1 for initial blocks, I mean before the fork limit is -1 (unlimited) and after fork, we will put some number.

[the remark this answers is not in the document]
Comment (JekaMas) [reply consensus/bor/rest.go:51]: We can have a hard coded value for that case.

```
func getCapacity(totalLimit int) int {
    if totalLimit < 1 {
        // initial blocks case
        return 512 // for example
    }

    totalLimit /= 3
    if totalLimit < 1 {
        return 512
    }
    
    return totalLimit
}
```

[the remark this answers is not in the document]
Comment (0xKrishna) [reply consensus/bor/rest.go]: Done

**Review by JekaMas — APPROVED** [review]
lgtm

**Comment by jdkanani** [comment]
@0xKrishna can you also fix testcases?

**Comment by 0xKrishna** [comment]
> @0xKrishna can you also fix testcases?

Taking a look now

File: consensus/bor/rest.go [hunk 2]
```diff
@@ -47,8 +47,14 @@ func NewHeimdallClient(urlString string) (*HeimdallClient, error) {
 	return h, nil
 }
 
-func (h *HeimdallClient) FetchStateSyncEvents(fromID uint64, to int64) ([]*EventRecordWithTime, error) {
-	eventRecords := make([]*EventRecordWithTime, 0)
+func (h *HeimdallClient) FetchStateSyncEvents(fromID uint64, to int64, totalLimit int) ([]*EventRecordWithTime, error) {
+	var capacity int
+	if limit := totalLimit / 3; limit >= 1 {
```
Comment (cffls) [inline consensus/bor/rest.go:52]: It may improve the readability by adding a comment here to briefly explain the purpose of `/ 3`, thanks!

**Review by ssandeep — APPROVED** [review]

**Comment by 0xKrishna** [comment]
Don't need HF for now, Will implement a better solution later.

**Pull request closed without merging**